### PR TITLE
moves `addUnique` to `std/sequtils`

### PR DESCRIPTION
**Pull request by jmgomez**

### CODE REVIEW

File: lib/pure/collections/sequtils.nim [hunk 1]
```diff
@@ -140,6 +140,31 @@ func concat*[T](seqs: varargs[seq[T]]): seq[T] =
       result[i] = itm
       inc(i)
 
+func addUnique*[T](s: var seq[T], x: sink T) =
```
Comment (juancarlospaco) [inline lib/pure/collections/sequtils.nim:143]: Should the same exist for `string` in `strutils` ?. 🤔

**Comment by Varriount** [comment]
What's wrong with `if x notin sequence: sequence.add(x)`? I don't feel that this function really merits being added to the standard library.

File: lib/pure/collections/sequtils.nim [hunk 2]
```diff
@@ -140,6 +140,31 @@ func concat*[T](seqs: varargs[seq[T]]): seq[T] =
...
+  ## Uses `==` to check if the item is already present.
+  runnableExamples:
+    var a = @[1, 2, 3]
+    a.addUnique(4)
+    a.addUnique(4)
+    assert a == @[1, 2, 3, 4]
+
+  for i in 0..high(s):
+    if s[i] == x: return
+  s.add move(x)
```
Comment (Araq) [inline lib/pure/collections/sequtils.nim]:
```suggestion
  s.add ensureMove(x)
```

File: lib/pure/collections/sequtils.nim (hunk 1 above)
Comment (Araq) [reply lib/pure/collections/sequtils.nim:143]: Can be added later.

File: lib/pure/collections/sequtils.nim [hunk 3]
```diff
@@ -140,6 +140,31 @@ func concat*[T](seqs: varargs[seq[T]]): seq[T] =
...
+  ## Uses `==` to check if the item is already present.
+  runnableExamples:
+    var a = @[1, 2, 3]
+    a.addUnique(4)
+    a.addUnique(4)
+    assert a == @[1, 2, 3, 4]
+
+  for i in 0..high(s):
+    if s[i] == x: return
+  s.add ensureMove(x)
```
Comment (Araq) [inline lib/pure/collections/sequtils.nim]:
```suggestion
  when declared(ensureMove):
    s.add ensureMove(x)
  else:
    s.add x
```

File: lib/pure/collections/sequtils.nim (hunk 1 above)
Comment (juancarlospaco) [reply lib/pure/collections/sequtils.nim:143]: Then this needs a follow up PR.

**Pull request merged**